client/src/components/addCenterComponent.js

```diff
@@ -0,0 +1,234 @@
+import React from 'react';
+import { ModalHeader } from './modalHeader';
+import { FormGroup } from './formGroup';
+import centerNameIcon from '../resources/images/glyphicons-21-home.png';
+import centerImageIcon from '../resources/images/glyphicons-139-picture.png';
+import addressIcon from '../resources/images/glyphicons-243-map-marker.png';
+import cityIcon from '../resources/images/glyphicons-740-cadastral-map.png';
+import capacityIcon from '../resources/images/glyphicons-44-group.png';
+import amountIcon from '../resources/images/glyphicons-228-usd.png';
+import { Option } from './selectOption';
+import axios from 'axios';
+import { logout, apiLink } from '../reusables';
+import { connect } from 'react-redux';
+import { setCenterDetails } from '../actions/centerActions';
+
+const inputAttrs = (inputType, inputName, placeholder, className, ref, required) => ({
+  inputType,
+  inputName,
+  placeholder,
+  className,
+  ref,
+  required,
+});
+
+const mapDispatchToProps = dispatch => ({
+  setCenterDetails: center => dispatch(setCenterDetails(center)),
+  // setCenterDefaults: data => dispatch(setCenterDefaults(data)),
+});
+const mapStateToProps = state => ({
+  modalTitle: state.page.modalTitle,
+  required: state.page.required,
+  centerDefaults: state.page.centerDefaults,
+});
+let centerId;
+
+class AddCenterComponent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.folder = apiLink === 'http://localhost:8080' ? 'dev/centers' : 'prod/centers';
+  }
+  handleSubmit(event) {
```

Comment:
CRITICAL: I think this method is doing too much and is too long. You should consider breaking out smaller functions and calling those functions within this one.

NITPICK: You might also want to consider moving some of the actions / api logic into an `actions` file and import/call the functions e.g. `createCenter`, `updateCenter` here.